packages/boot/src/mixins/boot.mixin.ts

```diff
@@ -0,0 +1,152 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: @loopback/boot
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {discoverArtifactFiles} from '../utils';
+import {resolve} from 'path';
+import {Constructor} from '@loopback/context';
+import {StringOrArr} from '../types';
+const debug = require('debug')('@loopback/boot:mixin');
+
+/**
+ * A mixin class for Application that boots it. Boot automatically
+ * binds controllers based on convention based naming before starting
+ * the Application.
+ *
+ * ```ts
+ * class MyApp extends BootMixin(Application) {}
+ * ```
+ */
+// tslint:disable-next-line:no-any
+export function BootMixin<T extends Constructor<any>>(superClass: T) {
+  return class extends superClass {
+    // tslint:disable-next-line:no-any
+    constructor(...args: any[]) {
+      super(...args);
+
+      // Create Options Object / Init Defaults if non-existent!
+      if (!this.options) this.options = {};
+      if (!this.options.boot) {
+        this.options.boot = {};
+      }
+
+      // Set `dist/src` as rootDir (as a relative path) if not set by user
+      if (!this.options.boot.rootDir) {
+        this.options.boot.rootDir = 'dist/src';
+      }
+
+      // Resolve to Absolute path! This is done here so user can provide
+      // relative path or absolute path for rootDir.
+      this.options.boot.rootDir = resolve(this.options.boot.rootDir);
+    }
+
+    /**
+     * Default function to boot Application. Boot is the process of discovering
+     * and binding artifacts automatically to a given key.
+     */
+    async boot() {
+      this.bootControllers();
+      // This allows boot to work regardless of Mixin order
+      if (super.boot) await super.boot();
+    }
+
+    /**
+     * Override the start function to boot before running start!
+     */
+    async start() {
+      await this.boot();
+      super.start();
+    }
+
+    /**
+     * Boot controllers automatically by finding them in given directories
+     * and binding them using this.controller() function.
+     */
+    bootControllers() {
+      // Default folder to look in (relative to rootDir)
+      const controllerDirs = this.options.boot.controllerDirs || 'controllers';
```

Comment:
Please export the default value as a constant (for your unit tests, and for the convenience of other devs).